utilities/transactions/pessimistic_transaction.h

```diff
@@ -111,6 +110,12 @@ class PessimisticTxn : public TransactionBaseImpl {
   int64_t GetDeadlockDetectDepth() const { return deadlock_detect_depth_; }
 
  protected:
+  virtual Status PrepareInternal() = 0;
+
+  virtual Status CommitSingleInternal() = 0;
```

Comment:
What does commit_single mean? Seems it means a non-2PC transaction, without any consistency checking? Add comment here to explain.

And if the transaction API allow someone call Commit() without Prepare(), what if someone mean to use 2PC, but forget to call Prepare()..